storm-core/src/jvm/org/apache/storm/daemon/supervisor/SyncSupervisorEvent.java

```diff
@@ -0,0 +1,631 @@
+        Map stormConf = Utils.fromCompressedJsonConf(FileUtils.readFileToByteArray(new File(stormconfPath)));
+        Map<String, Map<String, Object>> blobstoreMap = (Map<String, Map<String, Object>>) stormConf.get(Config.TOPOLOGY_BLOBSTORE_MAP);
+        List<String> blobFileNames = new ArrayList<>();
+        if (blobstoreMap != null) {
+            for (Map.Entry<String, Map<String, Object>> entry : blobstoreMap.entrySet()) {
+                String key = entry.getKey();
+                Map<String, Object> blobInfo = entry.getValue();
+                String ret = null;
+                if (blobInfo != null && blobInfo.containsKey("localname")) {
+                    ret = (String) blobInfo.get("localname");
+                } else {
+                    ret = key;
+                }
+                blobFileNames.add(ret);
+            }
+        }
+        for (String string : blobFileNames) {
+            if (!Utils.checkFileExists(string))
+                return false;
+        }
+        return true;
+    }
+
+    /**
+     * Download all blobs listed in the topology configuration for a given topology.
+     * 
+     * @param conf
+     * @param stormconfPath
+     * @param localizer
+     * @param tmpRoot
+     */
+    protected void downloadBlobsForTopology(Map conf, String stormconfPath, Localizer localizer, String tmpRoot) throws IOException {
+        Map stormConf = ConfigUtils.readSupervisorStormConfGivenPath(conf, stormconfPath);
+        Map<String, Map<String, Object>> blobstoreMap = (Map<String, Map<String, Object>>) stormConf.get(Config.TOPOLOGY_BLOBSTORE_MAP);
+        String user = (String) stormConf.get(Config.TOPOLOGY_SUBMITTER_USER);
+        String topoName = (String) stormConf.get(Config.TOPOLOGY_NAME);
+        File userDir = localizer.getLocalUserFileCacheDir(user);
+        List<LocalResource> localResourceList = SupervisorUtils.blobstoreMapToLocalresources(blobstoreMap);
+        if (localResourceList.size() > 0) {
+            if (!userDir.exists()) {
+                FileUtils.forceMkdir(userDir);
+            }
+            try {
+                List<LocalizedResource> localizedResources = localizer.getBlobs(localResourceList, user, topoName, userDir);
+                setupBlobPermission(conf, user, userDir.toString());
+                for (LocalizedResource localizedResource : localizedResources) {
+                    File rsrcFilePath = new File(localizedResource.getFilePath());
+                    String keyName = rsrcFilePath.getName();
+                    String blobSymlinkTargetName = new File(localizedResource.getCurrentSymlinkPath()).getName();
+
+                    String symlinkName = null;
+                    if (blobstoreMap != null) {
+                        Map<String, Object> blobInfo = blobstoreMap.get(keyName);
+                        if (blobInfo != null && blobInfo.containsKey("localname")) {
```

Comment:
this logic was wrapped inside get-blob-localname in clojure. In java, it is duplicated in three places.